# Case Study on Testing of Hypothesis

In [1]:
# importing pandas as pd
import pandas as pd

#Read dataset 
Data=pd.read_csv(r"Sales_add.csv")
Data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [2]:
Data.isnull().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

# A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.
**●The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.**

# Hypothesis:

# 
**Null hypothesis (H0):There is no increase in sales after stepping into digital marketing.**
<br>
**Alternative hypothesis (HA):There is an increase in sales after stepping into digital marketing.**
</p>

In [3]:
Data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [4]:
from scipy import stats
#To conduct the paired sample t-test, one needs to use the stats.ttest_rel() method.
stats.ttest_rel(Data['Sales_before_digital_add(in $)'], Data['Sales_After_digital_add(in $)'])

Ttest_relResult(statistic=-12.09070525287017, pvalue=6.336667004575778e-11)

# INTERPRETATION OF THE RESULTS
**A paired sample t-test was used to analyze whether there is any increase in sales after stepping into digital marketing. if there is any increase in sales had a significant affect on after stepping into digital marketing. The sales before the digital marketing was lesser (149239.95 ± 14844.04 units) compared to the sales after the digital marketing (231123.43 ± 25556.78).** 
<br>
**Since the p-value of 6.37 is greater than α = 0.05, it would fail to reject the null hypothesis.**
<br>
<br>
**RESULT**
<br>
**There is an increase in sales after stepping into digital marketing.**
</p>

# ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

# Hypothesis:

# 
**H₀: whether there is no dependency between the features “Region” and “Manager”, i.e, dependent**
<br>
<br>
**H₁: whether there is any dependency between the features “Region” and “Manager” i,e, independent**
<br>
<br>
**α = 0.05**
<br>
<br>
**Following χ2 independence test statistics:**

In [5]:
# create contingency table
data_crosstab = pd.crosstab(Data['Region'],
                            Data['Manager'],
                           margins=True, margins_name="Total")

# significance level
alpha = 0.05

In [6]:
# Calcualtion of Chisquare test statistics
chi_square = 0
rows = Data['Region'].unique()
columns = Data['Manager'].unique()
for i in columns:
    for j in rows:
        O = data_crosstab[i][j]
        E = data_crosstab[i]['Total'] * data_crosstab['Total'][j] / data_crosstab['Total']['Total']
        chi_square += (O-E)**2/E

In [7]:
# The p-value approach
print(" The p-value approach to hypothesis testing in the decision rule")
p_value = 1 - stats.norm.cdf(chi_square, (len(rows)-1)*(len(columns)-1))
conclusion = "Failed to reject the null hypothesis."
if p_value <= alpha:
    conclusion = "Null Hypothesis is rejected."
        
print("chisquare-score is:", chi_square, " and p value is:", p_value)
print(conclusion)

 The p-value approach to hypothesis testing in the decision rule
chisquare-score is: 3.050566893424036  and p value is: 0.8287998105017607
Failed to reject the null hypothesis.


# RESULT:
**The features “Region” and “Manager”are independent.**    